### PR TITLE
deno-slack-sdk publish to npm on tag push

### PR DESCRIPTION
**Pull request by neptunel**
###  Summary

[NOTE]: This branch contains irrelevant commit history. For simplicity, closing this PR and noting that we've gotten approvals to merge the relevant change (build_npm.ts, github actions).

- Added the` npm_publish.yml` Git Action workflow that will auto-publish the npm build to NPM, so that Bolt node apps can consume the `deno-slack-sdk` with expanded type support.

- The push event only being triggered when there's a tag pushed in the `number.number.number`(ex: 0.5.0) format to follow the semantic versioning. 

- NPM_TOKEN will be added to `deno-slack-sdk ` repo's environment secrets using Slack API's own npm token after this PR reviewed and merged.
### Requirements (place an `x` in each `[ ]`)

* [ x ] I've read and understood the [Contributing Guidelines](https://github.com/{project_slug}/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [ x ] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: build_npm.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  outDir: "./npm",
+  shims: {
+    // see JS docs for overview and more options
+    deno: true,
+  },
+  package: {
+    // package.json properties
+    name: "@slack/deno-slack-sdk",
+    version: Deno.args[0],
+    description: "Test deno sdk dpendencies",
```
Comment (srajiang) [inline build_npm.ts:17]: Can we bring back the more helpful description? Think it might have gotten lost.

File: build_npm.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      url: "git+https://github.com/slackapi/deno-slack-sdk.git",
+    },
+    bugs: {
+      url: "https://github.com/slackapi/deno-slack-sdk/issues",
+    },
+  },
+});
+
+// post build steps
+// Deno.copyFileSync("LICENSE", "npm/LICENSE");
```
Comment (srajiang) [inline build_npm.ts:30]: Do we need this commented out line?

**Review by srajiang — CHANGES_REQUESTED** [review]

File: build_npm.ts (hunk 1 above)
Comment (neptunel) [reply build_npm.ts]: Done!

File: build_npm.ts (hunk 2 above)
Comment (neptunel) [reply build_npm.ts]: Removed this line!

**Review by srajiang — COMMENTED** [review]
Can we make sure that a /npm is included in .gitignore file? So that people don't accidentally commit up the build contents?

File: .github/workflows/npm-publish.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: Build & Deploy to NPM
+
+on:
+  push:    
+    tags:        
+      - '*.*.*'
```
Comment (srajiang) [inline .github/workflows/npm-publish.yml:9]: Nice job constraining this down to tags that follow semantic versioning

**Review by srajiang — APPROVED** [review]
Have tested a: 

* Publishing a dummy package from sena's project fork via her test npm account as well as on the main `slackhq` account. We've cleaned up those packages.
* Tested that dummy version can be imported properly as a module in the main Bolt project.

**Review by stevengill — APPROVED** [review]

**Review by shapirone — APPROVED** [review]

**Pull request closed without merging**